delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/BlazegraphViewsQuery.scala

```diff
@@ -112,10 +213,11 @@ object BlazegraphViewsQuery {
     final case class VisitedAggregatedView(ref: ViewRef)                               extends VisitedView
   }
 
-  private[blazegraph] type FetchView = (IdSegment, ProjectRef) => IO[BlazegraphViewRejection, ViewResource]
+  private[blazegraph] type FetchView    = (IdSegment, ProjectRef) => IO[BlazegraphViewRejection, ViewResource]
+  private[blazegraph] type FetchProject = ProjectRef => IO[BlazegraphViewRejection, Project]
 
-  final def apply(acls: Acls, views: BlazegraphViews, client: BlazegraphClient)(implicit
+  final def apply(acls: Acls, views: BlazegraphViews, projects: Projects, client: BlazegraphClient)(implicit
       config: ExternalIndexingConfig
   ): BlazegraphViewsQuery =
-    new BlazegraphViewsQueryImpl(views.fetch, acls, client)
+    new BlazegraphViewsQueryImpl(views.fetch, pRef => projects.fetchProject(pRef), acls, client)
```

Comment:
`projects.fetchProject` should work
or `projects.fetchProject[ProjectNotFound]`